internal/actions/update.go

```diff
@@ -34,7 +34,7 @@ func Update(client container.Client, params types.UpdateParams) (types.Report, e
 
 	for i, targetContainer := range containers {
 		stale, newestImage, err := client.IsContainerStale(targetContainer)
-		shouldUpdate := stale && !params.NoRestart && !params.MonitorOnly && !targetContainer.IsMonitorOnly()
+		shouldUpdate := stale && !params.NoRestart && (( !params.MonitorOnly && !targetContainer.IsMonitorOnly() ) || ( params.LabelPrecedence && !targetContainer.IsMonitorOnly() ))
```

Comment:
I hated this line before and this makes the problem even worse... We have to be able to do this in a more legible way...

Reply:
Perhaps we can have a `containerMonitorOnly` that just deals with resolving whether a container has `monitor-only` enabled or not?

Reply:
Okay, my suggestion is this: 
In the `IsEnabled` and `IsMonitorOnly` functions for `container` (the go struct that abstracts the `docker inspect` raw data about a container), we can add a `GlobalConfig` which includes the argument settings for `labelPrecendence`, `monitorOnly`, etc. 
That way we can keep the logic inside the `container` funcs,  and making this statement much more readable. What do you think? I can help with this during the weekend.

Reply:
...actually, we could just pass the `params` to the functions, since that is already basically the `GlobalConfig`. 😅

Reply:
So i passed params to isMonitorOnly and the check is done inside.
(I initialy tried a containerMonitorOnly like varible before should update but it was even worse to read as the logic gets more complex)

PS: My test was wrong, I hope I fixed it correctly
PS2: I also fixed (I think) the previous test. At line 181, it was `_, err := actions.Update(client, types.UpdateParams{MonitorOnly: true})` and the check is `Expect(client.TestData.TriedToRemoveImageCount).To(Equal(0))` 
As there was not Cleanup: true in UpdateParams, for me there was no change that TriedToRemoveImageCount could be something else than 0

Reply:
As we discovered in #1753 that there is undocumented no-pull label, there is another change to do get expected behavior when label-take-precedence is used. Actual behavior is to not pull if either no-pull is set as argument or as label

Reply:
Actualy, no-pull label is documented in main, this option has been added un march

Reply:
I mark the conversation as resolved. label-take-precendence now also handle no-pull option